### PR TITLE
HttpBody mesage feature for stream RPC

### PR DESCRIPTION
**Pull request by adasari**
Hi,

HttpBody message Feature is not working with stream. the changes include fix the behavior

This possibly fix https://github.com/grpc-ecosystem/grpc-gateway/issues/1254

### CODE REVIEW

**Comment by adasari** [comment]
Didn't add changes to send the content type from the http body chunk.

File: runtime/handler.go [hunk 1]
```diff
@@ -66,12 +67,16 @@ func ForwardResponseStream(ctx context.Context, mux *ServeMux, marshaler Marshal
 		case resp == nil:
 			buf, err = marshaler.Marshal(errorChunk(streamError(ctx, mux.streamErrorHandler, errEmptyResponse)))
 		default:
-			result := map[string]interface{}{"result": resp}
-			if rb, ok := resp.(responseBody); ok {
-				result["result"] = rb.XXX_ResponseBody()
+			if httpBody, ok := resp.(*httpbody.HttpBody); ok {
+				buf, err = marshaler.Marshal(httpBody)
```
Comment (johanbrandhorst) [inline runtime/handler.go:71]: This shouldn't even need to consult the marshaler.
```suggestion
				buf = httpBody.GetData()
```

Comment (johanbrandhorst) [reply runtime/handler.go:71]: Probably want to do something about the delimiter too.

Comment (johanbrandhorst) [reply runtime/handler.go:71]: Please set the content type as well.

File: examples/internal/proto/examplepb/stream.swagger.json [hunk 2]
```diff
@@ -157,6 +188,28 @@
...
...ented as JSON, such as raw binary or\nan HTML page.\n\n\nThis message can be used both in streaming and non-streaming API methods in\nthe request as well as the response.\n\nIt can be used as a top-level request field, which is convenient if one\nwants to extract parameters from either the URL or HTTP template into the\nrequest fields and also want access to the raw HTTP body.\n\nExample:\n\n    message GetResourceRequest {\n      // A unique request id.\n      string request_id = 1;\n\n      // The raw HTTP body is bound to this field.\n      google.api.HttpBody http_body = 2;\n    }\n\n    service ResourceService {\n      rpc GetResource(GetResourceRequest) returns (google.api.HttpBody);\n      rpc UpdateResource(google.api.HttpBody) returns\n      (google.protobuf.Empty);\n    }\n\nExample with streaming methods:\n\n    service CaldavService {\n      rpc GetCalendar(stream google.api.HttpBody)\n        returns (stream google.api.HttpBody);\n      rpc UpdateCalendar(stream google.api.HttpBody)\n        returns (stream google.api.HttpBody);\n    }\n\nUse of this type only changes how the request and response bodies are\nhandled, all other features will continue to work unchanged."
```
Comment (johanbrandhorst) [inline examples/internal/proto/examplepb/stream.swagger.json:211]: This is obviously wrong, but it's probably not worth fixing right now. Could you raise an issue that the HttpBody swagger representation is wrong?

File: examples/internal/proto/examplepb/stream.swagger.json [hunk 3]
```diff
@@ -116,6 +116,37 @@
...
+            "schema": {
+              "type": "object",
+              "properties": {
+                "result": {
+                  "$ref": "#/definitions/apiHttpBody"
+                },
+                "error": {
+                  "$ref": "#/definitions/runtimeStreamError"
+                }
+              },
```
Comment (johanbrandhorst) [inline examples/internal/proto/examplepb/stream.swagger.json:135]: In this case we'll need to override the swagger schema, but again, lets do that in a separate issue.

File: runtime/handler.go (hunk 1 above)
Comment (adasari) [reply runtime/handler.go:71]: will check on the content type.

File: examples/internal/proto/examplepb/stream.swagger.json (hunk 2 above)
Comment (adasari) [reply examples/internal/proto/examplepb/stream.swagger.json:211]: Sure.

**Comment by johanbrandhorst** [comment]
Closing this since we have https://github.com/grpc-ecosystem/grpc-gateway/pull/1273 against v2 instead

**Pull request closed without merging**